README.md

```diff
@@ -1 +1,794 @@
+- The nginx-example configuration requires special attention for SSL setup
+- SSL configuration depends on certificates that can only be created after enabling HTTP
+- Recommended approach: Create certificates → Enable SSL
+
+## Overview
+This repository provides a comprehensive infrastructure setup using Docker Compose to deploy various services including:
+- Jitsi (Video Conferencing)
+- WordPress (Content Management)
+- Nextcloud (File Sharing & Collaboration)
+- Monitoring Stack (Loki, Prometheus, Grafana)
+- Additional utilities (Watchtower, Jumphost)
+
+The goal is to demonstrate Infrastructure as Code practices while providing a lightweight, maintainable solution for self-hosted services.
+
+## Prerequisites
+
+### System Requirements
+- Linux server (tested with Ubuntu 24.04)
+- Internet domain with DNS control
+- Docker Compose v2.33.1 or later
+- Open firewall ports (see Firewall Configuration)
+
+### Required Docker Plugins
+- Loki logging plugin (required for container logging):
+```bash
+docker plugin install grafana/loki-docker-driver:latest --alias loki --grant-all-permissions
+```
+
+### Domain Configuration
+Ensure your domain's DNS settings are properly configured to point to your server's IP address.
+
+## Installation
+
+### Docker Setup
+1. Install Docker and Docker Compose:
+```bash
+sudo apt install docker docker-compose
+```
+
+2. Add your user to the docker group (to run Docker without sudo):
+```bash
+sudo usermod -aG docker $USER
+```
+
+### Firewall Configuration
+1. Install and configure UFW:
+```bash
+sudo apt install ufw
+sudo ufw allow 22/tcp
+sudo ufw enable
+```
+
+2. Verify firewall status:
+```bash
+sudo ufw status
+```
+
+### Fail2ban Setup (Recommended)
+1. Install fail2ban:
+```bash
+sudo apt install fail2ban
+sudo cp /etc/fail2ban/jail.conf /etc/fail2ban/jail.local
+```
+
+2. Configure fail2ban (edit `/etc/fail2ban/jail.local`):
+```ini
+[sshd]
+enabled = true 
+port    = ssh
+filter = sshd[mode=aggressive]
+logpath = %(sshd_log)s
+backend = %(sshd_backend)s
+maxretry = 3 
+findtime = 3600
+bantime = 1200
+```
+
+3. Enable and start fail2ban:
+```bash
+sudo systemctl enable fail2ban 
+sudo systemctl start fail2ban
+```
+
+4. Verify fail2ban status:
+```bash
+systemctl status fail2ban
+fail2ban-client status sshd
+```
+
+> **Security Note**: It's highly recommended to disable password authentication for SSH and use only public key authentication.
+
+# Jitsi and Reverse Proxy Setup
+
+This section describes how to set up Jitsi Meet with a reverse proxy for secure access. The setup consists of two main components:
+1. Nginx reverse proxy with SSL termination
+2. Jitsi Meet stack (Prosody, Jicofo, JVB, and Web interface)
+
+## Prerequisites
+- Docker and Docker Compose installed
+- A domain name pointing to your server
+- Ports 80 and 443 available on your server
+- Environment variables configured (see `env.example` files in both `proxy` and `jitsi` directories)
+
```

Comment:
One still needs to source the environment file, or?
So something like "source ./proxy/env.example" and "source ./jitsi/env.example".